packages/block-library/src/search/index.php

```diff
@@ -131,9 +131,23 @@ function render_block_core_search( $attributes ) {
 				$button_internal_markup = wp_kses_post( $attributes['buttonText'] );
 			}
 		} else {
-			$button_classes[]       = 'has-icon';
+			$button_classes[] = 'has-icon';
+			$icon_dimensions  = '24';
+			if ( ! empty( $attributes['style']['typography']['fontSize'] ) ) {
+				$icon_dimensions = $attributes['style']['typography']['fontSize'];
+			} elseif ( ! empty( $attributes['fontSize'] ) ) {
+				$font_size_presets = wp_get_global_settings( array( 'typography', 'fontSizes' ) );
+				if ( isset( $font_size_presets['theme'] ) || isset( $font_size_presets['default'] ) ) {
+					$font_size_presets     = $font_size_presets['theme'] ?? $font_size_presets['default'];
```

Comment:
I think this should be fine. Looks like presets are always keyed by origin?

https://github.com/WordPress/gutenberg/blob/f1b41e5dd4fb1e327eb890c3cf2fda289e53abfd/lib/class-wp-theme-json-gutenberg.php#L722-L722

Reply:
I haven't tested this out, but if a theme sets its own font sizes and also uses the default font sizes (introduced in https://github.com/WordPress/gutenberg/pull/56661), what happens if someone sets the Search block to one of the default font sizes? Do we need an array with all the combined presets, I wonder 🤔

Reply:
> Do we need an array with all the combined presets, I wonder 

This is a good point - I'm currently looking into it. 

Not yet sure if https://github.com/WordPress/gutenberg/issues/57733 needs to be fixed first. 🤔

Reply:
@andrewserong

#56661 was reverted in https://github.com/WordPress/gutenberg/pull/58456

 I this PR loosely matches what Gutenberg is doing, with the exception of `custom`, which I'll add.

Font sizes presets are collated for the font size picker:

https://github.com/WordPress/gutenberg/blob/7f97bcc7506afa4973e9eeff4b3909ca88bd6b75/packages/block-editor/src/components/global-styles/typography-panel.js#L107-L107

With the following prioritization:

https://github.com/WordPress/gutenberg/blob/7f97bcc7506afa4973e9eeff4b3909ca88bd6b75/packages/block-editor/src/store/get-block-settings.js#L120-L122

So no merging... yet.

Still, looking at https://github.com/WordPress/gutenberg/issues/57733 the backend still doesn't have the equivalent of `getUniqueFontSizesBySlug`.

I'm starting to believe this warrants a new function in `typography.php`, something that does the deduping, but also can do the merging later (similar to the [reverted JS function)](https://github.com/WordPress/gutenberg/pull/58456/files#diff-def0802a58e1512057e07b22af00a628023b325ea4bdf28a87b322529405cdf2L120) should it ever be revisited. 

cc @ajlende in case I'm missing context

Reply:
Oh, thank you, I didn't realise it'd been reverted!